### PR TITLE
pkp/pkp-lib#1537 consider public URL identifier in issue required policy

### PR DESCRIPTION
**Pull request by bozana**
S. https://github.com/pkp/pkp-lib/issues/1537

### CODE REVIEW

File: classes/security/authorization/OjsIssueRequiredPolicy.inc.php [hunk 1]
```diff
@@ -64,6 +64,31 @@ function dataObjectEffect() {
...
+	/**
+	 * @copydoc DataObjectRequiredPolicy::getDataObjectId()
+	 * Considers a not numeric public URL identifier
+	 */
+	function getDataObjectId() {
+		// Identify the data object id.
+		$router = $this->_request->getRouter();
+		switch(true) {
+			case is_a($router, 'PKPPageRouter'):
+				if ( is_numeric($this->_request->getUserVar($this->_parameterName)) ) {
```
Comment (asmecher) [inline classes/security/authorization/OjsIssueRequiredPolicy.inc.php]: Might be better to use `ctype_digit` as in [this example from the stable codebase](https://github.com/pkp/ojs/blob/ojs-stable-2_4_8/classes/issue/IssueDAO.inc.php#L183). `is_numeric` will return true with e.g. `0xa`, `2e3`, and other unexpected notations.

**Comment by bozana** [comment]
@asmecher, I change it, s. the new commit. I will then change it also in the basis class in the pkp-lib and make a PR, ok?

File: classes/security/authorization/OjsIssueRequiredPolicy.inc.php [hunk 2]
```diff
@@ -74,7 +74,7 @@ function getDataObjectId() {
 		$router = $this->_request->getRouter();
 		switch(true) {
 			case is_a($router, 'PKPPageRouter'):
-				if ( is_numeric($this->_request->getUserVar($this->_parameterName)) ) {
+				if ( ctype_digit($this->_request->getUserVar($this->_parameterName)) ) {
```
Comment (asmecher) [inline classes/security/authorization/OjsIssueRequiredPolicy.inc.php]: Add a string cast -- `ctype_digit(123)` returns false, but `ctype_digit("123")` returns true.

**Comment by bozana** [comment]
Like this?

**Comment by asmecher** [comment]
Yes, that should do it!

**Comment by bozana** [comment]
OK, I will then also make a PR for the basis class in pkp-lib... THANKS!!!

**Pull request merged**